src/Converter.php

```diff
@@ -314,7 +314,7 @@ protected function getEntityRanges(string $text, array $characterMetaList): arra
         $ranges = [];
         $rangeStart = 0;
 
-        foreach (str_split($text) as $i => $char) {
+        foreach ($this->str_split_utf8($text) as $i => $char) {
```

Comment:
Using `mb_str_split` didn't do the job?

```suggestion
        foreach (mb_str_split($text) as $i => $char) {
```